scripts/generate-json-schema.ts

```
@@ -205,11 +215,40 @@ async function generateJsonSchema() {
 		}),
-		// Write schema files
-		Result.map(schema => JSON.stringify(schema, null, '\t')),
 		Result.unwrap(),
 	);
 
+	// Check if existing root schema is identical to avoid unnecessary writes
+	const existingRootSchema = await Result.pipe(
+		readFile(SCHEMA_FILENAME),
+		Result.map(content => JSON.parse(content) as unknown),
+		Result.unwrap(''),
+	);
```

ESLint

[error] Unsafe assignment of an error typed value.

(ts/no-unsafe-assignment)